datafusion/common/src/scalar.rs

```diff
@@ -4486,4 +4922,623 @@ mod tests {
             assert!(distance.is_none());
         }
     }
+
+    #[test]
+    fn test_scalar_interval_add() {
+        let cases = [
+            (
+                ScalarValue::IntervalYearMonth(Some(IntervalYearMonthType::make_value(
```

Comment:
I think you can use https://docs.rs/datafusion/20.0.0/datafusion/scalar/enum.ScalarValue.html#method.new_interval_ym to reduce this boilerplate substantially

```suggestion
                ScalarValue::new_iterval_ym(1, 12)
```

The same for most of the other tests

Reply:
Done